keras_core/backend/jax/distribution.py

```diff
@@ -0,0 +1,115 @@
+"""!!!DO NOT USE!!!
+
+Distribution related class for JAX backend.
+
+This is just a prototype and we might want to unify it in future for other
+backends.
+"""
+
+import contextlib
+
+from absl import logging
+import jax
+import numpy as np
+
+from keras_core.backend.common import global_state
+
+DEFAULT_BATCH_DIM_NAME = "batch"
+GLOBAL_ATTRIBUTE_NAME = "distribution"
+
+
+def get_distribution():
+    """Retrieve the current distribution from global context."""
+    return global_state.get_global_attribute(GLOBAL_ATTRIBUTE_NAME)
+
+
+class DataParallelDistribution:
+    def __init__(self, mesh=None, devices=None):
+        """Create the data parallel distribution.
+
+        User can choose to create this instance by either `Mesh` or `devices`
+        parameters (but not both).
+
+        The mesh is expected to be a `jax.sharding.Mesh` instance, and is
+        expected to be 1D only. In case that the mesh has multiple axises, then
+        the first axis will be treated as data parallel dimension (and a warning
+        will be raised).
+
+        When a list of `devices` are provided, they will be used to construct a
+        1D mesh.
+
+        When both `mesh` and `devices` are absent, then we will rely on
+        `jax.devices` to detect any available devices, and create mesh from
+        them.
+        """
+        super().__init__()
+        if mesh:
+            self._initialize_with_mesh(mesh)
+        elif devices:
+            self._initialize_mesh_from_devices(devices)
+        else:
+            self._initialize_mesh_from_jax_devices()
+
+        self._configure_sharding_spec()
+        self._batch_dim_name = self.mesh.axis_names[0]
+
+    @contextlib.contextmanager
+    def scope(self):
+        original_scope = global_state.get_global_attribute(
+            GLOBAL_ATTRIBUTE_NAME
+        )
+        global_state.set_global_attribute(GLOBAL_ATTRIBUTE_NAME, self)
+        try:
+            yield
+        finally:
+            global_state.set_global_attribute(
+                GLOBAL_ATTRIBUTE_NAME, original_scope
+            )
+
+    def as_global_distribution(self):
```

Comment:
For consistency with the Dtype Policy API, I think we should make this into a `set_distribution()` standalone function.

Reply:
Done. Added `set_global_distribution()` and rename `get_distribution` to `get_global_distribution`

Reply:
For strict consistency with other global config APIs (`image_data_format`, `dtype_policy`, `floatx`) we'll need a round of renaming I think -- but we can handle that later.